clients/src/main/java/org/apache/kafka/common/metrics/stats/SampledStat.java

```diff
@@ -34,35 +35,56 @@
  */
 public abstract class SampledStat implements MeasurableStat {
 
-    private double initialValue;
+    private final double initialValue;
+    /**
+     * Index of the latest stored sample.
+     */
     private int current = 0;
+    /**
+     * Stores the recorded samples in a ring buffer.
```

Comment:
I found this a bit confusing. To me "ring buffer" means a fixed size list with mutable head and tail indices. But `samples` isn't like that. Rather it's a list of mutable samples, and `Sample.reset` is used to overwrite old samples. The effect is the same, of course, but I wonder if we could review the comment:

```suggestion
     * Stores the recorded samples. Older samples are overwritten using {@link Sample#reset(long)}.
```

Reply:
That sounds fair. I have fixed the java doc in the latest revision and replaced with the following
```
    /**
     * Stores the recorded samples.
     * Note that the previously recorded samples may be overwritten/reset if they are considered obsolete by the
     * {@link Sample#purgeObsoleteSamples} function.
     */
```